ambry-server/src/main/java/com.github.ambry.server/VCRServer.java

```diff
@@ -0,0 +1,240 @@
+import com.github.ambry.clustermap.DataNodeId;
+import com.github.ambry.clustermap.VirtualReplicatorCluster;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.ClusterMapConfig;
+import com.github.ambry.config.ConnectionPoolConfig;
+import com.github.ambry.config.NetworkConfig;
+import com.github.ambry.config.ReplicationConfig;
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.config.ServerConfig;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.network.BlockingChannelConnectionPool;
+import com.github.ambry.network.ConnectionPool;
+import com.github.ambry.network.NetworkServer;
+import com.github.ambry.network.Port;
+import com.github.ambry.network.PortType;
+import com.github.ambry.network.SocketServer;
+import com.github.ambry.notification.NotificationSystem;
+import com.github.ambry.store.FindTokenFactory;
+import com.github.ambry.store.StoreKeyConverterFactory;
+import com.github.ambry.store.StoreKeyFactory;
+import com.github.ambry.utils.SystemTime;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.utils.Utils.*;
+
+
+/**
+ * Virtual Cloud Replicator server
+ */
+public class VCRServer {
+
+  private CountDownLatch shutdownLatch = new CountDownLatch(1);
+  private NetworkServer networkServer = null;
+  private ScheduledExecutorService scheduler = null;
+  private CloudBackupManager cloudBackupManager = null;
+  private Logger logger = LoggerFactory.getLogger(getClass());
+  private final VerifiableProperties properties;
+  private final ClusterAgentsFactory clusterAgentsFactory;
+  private ClusterMap clusterMap;
+  private ClusterParticipant clusterParticipant;
+  private VirtualReplicatorCluster virtualReplicatorCluster;
+  private MetricRegistry registry = null;
+  private JmxReporter reporter = null;
+  private ConnectionPool connectionPool = null;
+  private final NotificationSystem notificationSystem;
+  private CloudDestinationFactory cloudDestinationFactory;
+  private SSLConfig sslConfig;
+  private VCRMetrics metrics = null;
+
+  /**
+   * VCRServer constructor.
+   * @param properties the config properties to use.
+   * @param clusterAgentsFactory the {@link ClusterAgentsFactory} to use.
+   * @param notificationSystem the {@link NotificationSystem} to use.
+   */
+  public VCRServer(VerifiableProperties properties, ClusterAgentsFactory clusterAgentsFactory,
+      NotificationSystem notificationSystem) {
+    this.properties = properties;
+    this.clusterAgentsFactory = clusterAgentsFactory;
+    this.notificationSystem = notificationSystem;
+  }
+
+  /** Test constructor */
```

Comment:
Looks like we don't need this any more after making VCR ssl option. 
If still needed, please add javadoc for arguments.

Reply:
Added javadoc